product_supplierinfo_import_by_barcode/tests/test_product_supplierinfo_import_by_barcode.py

```diff
@@ -0,0 +1,317 @@
+                ],
+            }
+        )
+        cls.import_template_2 = cls.env["product.supplierinfo.import.template"].create(
+            {
+                "name": "Olé Drinks S.A.",
+                "search_header_name": "EAN",
+                "header_offset": 1,
+                "template_line_ids": [
+                    Command.create(
+                        {
+                            "header_name": "Description",
+                            "field_id": cls.env.ref(
+                                "product.field_product_supplierinfo__product_name"
+                            ).id,
+                        }
+                    ),
+                    Command.create({"header_name": "Ref"}),
+                    Command.create({"header_name": "Price"}),
+                    Command.create({"header_name": "Disc\n%"}),
+                    Command.create({"header_name": "Prom\n%"}),
+                    Command.create(
+                        {
+                            "header_name": "Final",
+                            "field_id": cls.env.ref(
+                                "product.field_product_supplierinfo__price"
+                            ).id,
+                        }
+                    ),
+                ],
+            }
+        )
+        cls.prod_horchata = cls.env["product.product"].create(
+            {
+                "name": "Horchata",
+                "barcode": "000001",
+            }
+        )
+        cls.product_salmorejo = cls.env["product.product"].create(
+            {
+                "name": "Salmorejo",
+                "barcode": "000003",
+                "seller_ids": [
+                    Command.create(
+                        {
+                            "partner_id": cls.supplier_1.id,
+                            "date_start": "2024-01-01",
+                            "min_qty": 1,
+                            "price": 1.33,
+                        }
+                    )
+                ],
+            }
+        )
+        cls.salmorejo_initial_supplierinfo = cls.product_salmorejo.seller_ids
+
+    def _data_file(self, filename, encoding=None):
+        """Helper: load the excel file binary"""
+        mode = "rt" if encoding else "rb"
+        with open(path.join(path.dirname(__file__), filename), mode) as file:
+            data = file.read()
+            if encoding:
+                data = data.encode(encoding)
+            return b64encode(data)
+
+    def _import_supplierinfo_file(
+        self, file_path, supplier, delay=0, create_new_products=True
+    ):
+        """ "Helper: import an excel file for testing"""
+        return self.env["product.supplierinfo.import"].create(
+            {
+                "supplierinfo_filename": file_path,
+                "supplierinfo_file": self._data_file(file_path),
+                "supplier_id": supplier.id,
+                "date_start": fields.Date.today(),
```

Comment:
In business code, I prefer to use `context_today` to prevent mistakes with time zones. In test-related code, it's not very important, but it can cause tests to fail at certain hours if you use it in any assert. If you can update it to my suggestion, that would be great, but it's not blocking.
```suggestion
                "date_start": fields.Date.context_today(),
```